src/super_gradients/common/sg_loggers/base_sg_logger.py

```diff
@@ -329,6 +330,22 @@ def _save_checkpoint(self, path: str, state_dict: dict) -> None:
         if self.save_checkpoints_remote:
             self.model_checkpoints_data_interface.save_remote_checkpoints_file(self.experiment_name, self._local_dir, name)
 
+    @multi_process_safe
+    def add_yaml_summary(self, tag: str, summary_dict: dict, global_step: Optional[int] = None) -> None:
+        """Saves any dict to <experiment_folder>/<tag>.yaml
+        Initially added for saving metrics to yaml to store it in something easily parsable (easier than .pth checkpoints),
+        but who knows what it will be suited for later.
+
+        :param tag:         Identifier of the summary.
+        :param summary_dict:  Checkpoint summary_dict.
+        :param global_step: Epoch number.
+        """
+
+        name = tag + (f"_{global_step}" if global_step is not None else "") + ".yml"
+        with open(os.path.join(self._local_dir, name), "w") as outfile:
+            yaml.dump(summary_dict, outfile, default_flow_style=False)
+            outfile.close()
```

Comment:
No need to close file since it is already opened using `with` statement which guarantees close to be called.